src/ucs/datastruct/linear_func.h

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (C) Mellanox Technologies Ltd. 2019.  ALL RIGHTS RESERVED.
+ *
+ * See file LICENSE for terms.
+ */
+
+#ifndef UCS_LINEAR_FUNC_H_
+#define UCS_LINEAR_FUNC_H_
+
+#include <ucs/sys/compiler_def.h>
+
+
+/**
+ * A 1d linear function, represented as f(x) = c + x * m.
+ */
+typedef struct {
+    double         c;  /* constant factor */
+    double         m;  /* multiplicative factor */
+} ucs_linear_func_t;
+
+
+/**
+ * Calculate the linear function value for a specific point.
+ *
+ * @param [in] func    Linear function to apply.
+ * @param [in] x       Point to apply the function at.
+ *
+ * @return f(x)
+ */
+static UCS_F_ALWAYS_INLINE double
+ucs_linear_func_apply(const ucs_linear_func_t *func, double x)
+{
+    return func->c + (func->m * x);
+}
+
+
+/**
+ * Sum two linear functions.
+ *
+ * @param [out] result   Filled with the resulting linear function.
+ * @param [in]  func1    First function to add.
+ * @param [in]  funct2   Second function to add.
```

Comment:
func2

Reply:
thanks, fixed. force pushed the fixes since the PR is small, here is the [diff](https://github.com/openucx/ucx/compare/709b89e0f235b72580c0c5122e90bcad98a94659..580321551cef59d4177f9bf8367f1b6fa4896cde)

Reply:
> thanks, fixed. force pushed the fixes since the PR is small, here is the [diff](https://github.com/openucx/ucx/compare/709b89e0f235b72580c0c5122e90bcad98a94659..580321551cef59d4177f9bf8367f1b6fa4896cde)

@yosefe thank you 👍